tests/merch.test.ts

```diff
@@ -0,0 +1,100 @@
+import { ForbiddenError } from 'routing-controllers';
+import { UserAccessType } from '../types';
+import { DatabaseConnection, MerchFactory, UserFactory, PortalState } from './data';
+import { ControllerFactory } from './controllers';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('collection editing', () => {
+  test('Appropriate Permission Check for Getting Archived Items', async () => {
+    const conn = await DatabaseConnection.get();
+    const itemOption = MerchFactory.fakeOption();
+    const [item] = MerchFactory.itemsWith({
+      options: [itemOption],
+    });
+    const [collection] = MerchFactory.collectionsWith({
+      items: [item],
+      archived: false,
+    });
+
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [user] = UserFactory.with({ accessType: UserAccessType.STANDARD });
+
+    await new PortalState()
+      .createUsers([admin, user])
+      .createMerch([collection])
+      .write();
+
+    const merchStore = ControllerFactory.merchStore(conn);
+
+    const editCollection = {
+      collection: {
+        title: collection.title,
+        description: collection.description,
+        archived: true,
+      },
+    };
+
+    await merchStore.editMerchCollection({ uuid: collection.uuid }, editCollection, admin);
+
+    await expect(merchStore.getOneMerchCollection({ uuid: collection.uuid }, user)).rejects.toThrow(ForbiddenError);
+
+    const result = await merchStore.getOneMerchCollection({ uuid: collection.uuid }, admin);
+    expect(result.collection.uuid).toEqual(collection.uuid);
+  });
+
+  test('Ordering Archived Items is not allowed', async () => {
+    const conn = await DatabaseConnection.get();
+    const itemOption = MerchFactory.fakeOption();
```

Comment:
nit: there aren't necessarily "archived items" - there are "hidden" items that were set as hidden by a collection being archived, so "ordering hidden items" more accurately matches the behavior than "ordering archived items" 

also nit: capital casing is not needed